server_addon/houdini/client/ayon_houdini/plugins/publish/collect_cache_farm.py

```diff
@@ -27,47 +27,35 @@ def process(self, instance):
         # Why do we need this particular collector to collect the expected
         # output files from a ROP node. Don't we have a dedicated collector
         # for that yet?
+        # Answer: No, we don't have a generic expected file collector.
+        #         Because different product types needs different logic.
+        #         e.g. check CollectMantraROPRenderProducts
+        #               and CollectKarmaROPRenderProducts
         # Collect expected files
         ropnode = hou.node(instance.data["instance_node"])
         output_parm = lib.get_output_parameter(ropnode)
         expected_filepath = output_parm.eval()
         instance.data.setdefault("files", list())
         instance.data.setdefault("expectedFiles", list())
-        if instance.data.get("frames"):
-            files = self.get_files(instance, expected_filepath)
-            # list of files
-            instance.data["files"].extend(files)
-        else:
+
+        frames = instance.data.get("frames", "")
```

Comment:
This fallback doesn't seem right? We should either rely on `instance.data["frames"]` or do some explicit skipping like:

```suggestion
        frames = instance.data.get("frames")
        if frames is None:
            self.log.warning(f"No frames data found on instance {instance}. Skipping collection for caching on farm...")
```

Reply:
The fall back to empty string is important. 
without it the code will actually look similar to
```
"""
`CollectFrames` doesn't work with all product types in Houdini plugin.
e.g. hda
"""
frames = instance.data.get("frames")
if frames is None: # This means it's either has no files or has a single frame.
    # check file_path
    pass
elif isinstance(frames, str): # single file
    # use file_path directly
    # or combine staging directory and frames
    pass
elif isinstance(frames, list): # list of files
    # combine staging directory and each frame in frames
    pass
```
So, I combined it to 
```
frames = instance.data.get("frames", "")
if isinstance(frames, str): # single file
    # use file_path directly
    pass
else: # list of files
    # combine staging directory and each frame in frames
    pass
```

Reply:
For reference, I used this idea `frames = instance.data.get("frames", "")` 
in 
- collect_files_for_cleaning_up.py
- collect_cache_farm.py

Reply:
let me know if this commit makes more sense 
https://github.com/ynput/ayon-core/pull/9/commits/37cbcc13527741ad69d1f9c7c2b0145580df5825